lib/cloudregister/registerutils.py

```diff
@@ -509,6 +510,47 @@ def get_credentials(credentials_file):
     return (username, password)
 
 
+# ----------------------------------------------------------------------------
+def setup_registry(registry_fqdn, username, password):
+    set_registry_auth_token(registry_fqdn, username, password)
+
+
+# ----------------------------------------------------------------------------
+def set_registry_auth_token(registry_fqdn, username, password):
+    """Set the auth token to pull images from SUSE registry."""
+    auth_token = base64.b64encode('{username}:{password}'.format(
+        username=username,
+        password=password
+    ).encode()).decode()
+    registry_credentials = {}
+    registry_credentials[registry_fqdn] = {'auth': auth_token}
+
+    config_json = {}
+    try:
+        with open(REGISTRY_CREDENTIALS_PATH, 'r') as cred_json:
+            config_json = json.load(cred_json)
+        # file exists
+        # set the new registry credentials,
+        # independently of what that content was
+        config_json['auths'].update(registry_credentials)
+    except (FileNotFoundError, KeyError):
+        # config file does not exist or "auths" key is not set
+        os.makedirs(os.path.dirname(REGISTRY_CREDENTIALS_PATH), exist_ok=True)
+        config_json.update({'auths': registry_credentials})
+    except json.decoder.JSONDecodeError:
+        logging.info(
+            'Error found when opening %s' % REGISTRY_CREDENTIALS_PATH
+        )
+        config_json.update({'auths': registry_credentials})
+
+    with open(REGISTRY_CREDENTIALS_PATH, 'w') as cred_json_file:
```

Comment:
If opening the file for writing or the actual write failed the application will fail with a stack trace. Do we want to handle this error condition ?

Reply:
Yes, there shold be a try-except block here and further up the code where we handle file existence should be handled with `if os.path.exists(REGISTRY_CREDENTIALS_PATH):`